azure-toolkit-libs/azure-toolkit-auth-lib/src/main/java/com/microsoft/azure/toolkit/lib/auth/AzureAccount.java

```diff
@@ -5,38 +5,135 @@
 import java.util.List;
+import java.util.Map;
+import java.util.Objects;
 import java.util.Optional;
+import java.util.stream.Collectors;
 
 public class AzureAccount implements AzureService {
+    private final ClientLogger logger = new ClientLogger(AzureAccount.class);
 
     private Account account;
 
-    public List<Account> getAvailableAccounts() {
-        return null;
+    @Setter
+    @Getter
+    private AzureEnvironment environment;
+
+    public List<AccountEntity> getAvailableAccounts() {
+        Map<AuthType, IAccountEntityBuilder> allBuilders = buildProfilerBuilders(environment);
+        return allBuilders.values().stream().map(builder -> builder.build()).collect(Collectors.toList());
     }
 
-    /**
-     * @return account if authenticated
-     * @throws AzureToolkitAuthenticationException if failed to authenticate
-     */
-    public Account login(AccountEntity entity) throws AzureToolkitAuthenticationException {
-        final Account account = new Account(); // TODO: login and create Account instance;
-        this.account = account;
-        return this.account;
+    public void login(AuthConfiguration auth) {
+        environment = auth.getEnvironment();
+        boolean forceSPLogin = auth.getType() == AuthType.SERVICE_PRINCIPAL;
+        if (forceSPLogin || auth.getType() == AuthType.AUTO) {
+            boolean isSPConfigurationPresent = !StringUtils.isAllBlank(auth.getCertificate(), auth.getKey(),
+                    auth.getCertificatePassword());
+            try {
+                ValidationUtil.validateAuthConfiguration(auth);
+            } catch (InvalidConfigurationException e) {
+                if (forceSPLogin) {
+                    Lombok.sneakyThrow(e);
+                    return;
+                }
+                if (isSPConfigurationPresent) {
+                    logger.warning("Cannot login through 'SERVICE_PRINCIPAL' due to invalid configuration: " + e.getMessage());
+                }
+            }
+
+            ServicePrincipalAccountEntityBuilder builder = new ServicePrincipalAccountEntityBuilder(auth);
+            AccountEntity entity = builder.build();
+            if (entity.isAuthenticated()) {
+                login(entity);
+                return;
+            }
+        }
+        Map<AuthType, IAccountEntityBuilder> allBuilders = buildProfilerBuilders(environment);
+        if (allBuilders.containsKey(auth.getType())) {
+            loginByBuilder(allBuilders.get(auth.getType()));
+            return;
+        }
+
+        for (IAccountEntityBuilder builder : allBuilders.values()) {
+            if (loginByBuilder(builder)) {
+                return;
+            }
+        }
+    }
+
+    public void login(AccountEntity accountEntity) throws AzureToolkitAuthenticationException {
+        Objects.requireNonNull(accountEntity, "accountEntity is required.");
+        this.account = new Account();
+        final AccountEntity entity = new AccountEntity();
```

Comment:
create a copy constructor for `AccountEntity`?

Reply:
We cannot reuse the account entity because it is the inner state of Current AzureAccount => Account, otherwise the update on accountEnity passed to login will impact the the inner state AzureAccount